libdevcore/LevelDB.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "LevelDB.h"
+#include "Assertions.h"
+
+namespace dev
+{
+namespace db
+{
+namespace
+{
+
+inline leveldb::Slice toLDBSlice(Slice _slice)
+{
+    return leveldb::Slice(_slice.data(), _slice.size());
+}
+
+class LevelDBWriteBatch: public WriteBatchFace
+{
+public:
+	explicit LevelDBWriteBatch(leveldb::WriteOptions _writeOptions): m_writeOptions(std::move(_writeOptions))
+	{
+	}
+
+	void insert(Slice _key, Slice _value) override;
+	void kill(Slice _key) override;
+
+	const leveldb::WriteBatch& writeBatch() const { return m_writeBatch; }
+	leveldb::WriteBatch& writeBatch() { return m_writeBatch; }
+
+private:
+	const leveldb::WriteOptions m_writeOptions;
+	leveldb::WriteBatch m_writeBatch;
+};
+
+void LevelDBWriteBatch::insert(Slice _key, Slice _value)
+{
+	m_writeBatch.Put(toLDBSlice(_key), toLDBSlice(_value));
+}
+
+void LevelDBWriteBatch::kill(Slice _key)
+{
+	m_writeBatch.Delete(toLDBSlice(_key));
+}
+
+}
+
+leveldb::ReadOptions LevelDB::defaultReadOptions()
+{
+	leveldb::ReadOptions readOptions;
+	readOptions.verify_checksums = true;
```

Comment:
Did we have it on previously?

Reply:
Yes. I copied these options from the callers to leveldb, just wrapped them here.

Reply:
Actuallly I can't find where it was set to true, I think it was default `false`